src/slskd/Search/SearchService.cs

```diff
@@ -256,5 +261,30 @@ public bool TryCancel(Guid id)
 
             return false;
         }
+
+        /// <summary>
+        ///     Forces a cancel on the specified search.
+        /// </summary>
+        /// <param name="search">The search to force cancel.</param>
+        /// <returns>The operation context.</returns>
+        public async Task ForceCancel(Search search)
```

Comment:
after reviewing the existing search code i agree that this is necessary right now, but long term i'll create an issue to refactor searches so it isn't possible for them to fail and remain "stuck"